### ISSUE TITLE
Support for intermeddiate positions

### ISSUE THREAD

**Issue by cojoj** [tfohlmeister/homebridge-selve#5]
Hey! First of all, thank for this great plugin 🙏 It's o nice to be able to control Selve directly from the Home app ❤️

I was wondering whether it has crossed your mind to add support for the intermediate positions? While programming shutters you're able to set 2 positions for each shutter (not to mention stopping), and it's convenient to use those predefined settings from commeo remote. I'm not sure how this would work in HomeKit's reality, because Window Covering Service already has some predefined Characteristics, but from my understanding, an Accessory can have multiple Services ([example from Apple's website is a garage door with additional light](https://developer.apple.com/documentation/homekit/hmservice)). 

The obvious drawback would be that each Service would be represented as a different tile on Home's dashboard. Still, maybe there's a way of optionally configuring services so that user can decide whether he wants P1 and P2 buttons or not. 

I'll most likely try to play around with this and see what I can come up with as I don't have any previous experience with writing Homebridge plugins (or TS/JS in general). I'm creating this issue just to gather some info, confront my understanding of things and see if we're not duplicating our plans.

**Comment by tfohlmeister** [comment]
Hi and thanks for your comment!
I am not sure I understand though. You should be able to move the shutter to any position between 0 and 100% in steps of one.

In the Home app you have two ways to control each device: You can tap a tile which simply toggles it. Lamps turn off or on, shutters go up and down. But if you **long**-press or **"deep" press** (depending on your device) it opens up a detailed settings view, in which you can set additional options. For lamps that might be a color and brightness and for shutters you will get a vertical slider to fine adjust the position from 0 to 100%.

The same deep-press / long-press works in scenes and automation. First select the devices you want to include in your scene / automation, then fine-tune them.

Is this maybe something you missed or did I misunderstand your question completely?

**Comment by cojoj** [comment]
Looks like we're not on the same page, yet. You're right in everything you've said, and I'm aware of all of the **Force Tap gesture** etc. but the concept of intermediate positions is something else and I'll try to explain it better this time 😉

I'm using commeo Multi Send at home as well and when I was hooking it up with all of my shutters I was able to program two intermediate positions for each shutter (take a look at [manual](https://www.selve.de/fileadmin/documents/service/downloads/bedienungsanleitungen/funk/BA_commeo_Multi_Send_A5_4c_D_GB_F_NL_PL_web.pdf) p. 13 "7.6. Zwischenposition 1"), so this remote works like this:

+ when I tap up-arrow shutter will go all the way up
+ when I tap down it'll go all the way down
+ when I double tap up it'll go to intermediate position 1
+ when I double tap up it'll go to intermediate position 2
+ when I tap on the middle button motor will be paused at the current position

Right now in this plugin and with HomeKit Window Accessory you're able to achieve the first two points + additional position setting using the force tap gesture in which you can provide the desired value.

**Comment by tfohlmeister** [comment]
Sorry for getting back at you so late. I think I got you now!

You are missing an way to move a shutter to one of the predefined intermediate positions, correct?

You are right about an Accessory being able to expose multiple services. I could imagine adding an option to show two more virtual buttons maybe, which allow you to either move to position 1 or position 2? What do you think?
Checking the documentation though, I currently have no way to detect whether the shutter is currently at any intermediate position, so I can't automatically set the virtual button to *on* or *off*. But maybe that's not so important for a first version. 

I don't have my system setup with intermediate positions though, so you would have to test it! 😉

**Comment by cojoj** [comment]
@tfohlmeister I'd love to help as much as possible 😉 Yes, this is exactly what I'm referring to!

**Comment by tfohlmeister** [comment]
@cojoj Published a new version, please test! For config, see Readme or simply use the settings in homebridge-config-ui-x.

**Comment by tfohlmeister** [comment]
Just a quick heads-up @cojoj: I published a major update which requires changes in the config file. Please check the README for instructions on how to update your configuration.
Were you able to test out the new virtual switches for those intermediate positions?

**Comment by cojoj** [comment]
Thanks for the info @tfohlmeister! Yes, I was able to test intermediate positions and they seem to works fine! Also, I love the fact they're shown inside the shutter instead of being separate switches. Would be perfect if instead of a switch they'd be simple buttons, because right now switch sometimes stays on for some time etc. I wasn't able to find any info if there's actually such Service at all. Also, maybe adding some name to the positions would be a cool feature, because now on Homebridge's dashboard I have tons of Intermediate switches without any context 😂

One more thing is **stop button** functionality. I saw in your code that you've added XML for this functionality, but there's no service for this.

Here's the screenshot of how it looks like 😉 Thanks again for the very quick replies and implementation ❤️
![image](https://user-images.githubusercontent.com/4522407/88280916-057fe180-cce7-11ea-90f7-f98da0fea9d7.png)

**Comment by tfohlmeister** [comment]
Hi! Once again I am sorry for the late reply.

I just pushed out an update which fixes the main issues you encountered:
* virtual buttons now have a clearer name including the shutter name by default
* virtual buttons should now return to their **off** state more reliably
* a config option for a third virtual button to stop the shutter was added

Hope you like the update, let me know if you have any other ideas for improvements! 😀
